recipes/directx-headers/all/conanfile.py

```diff
@@ -0,0 +1,87 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rmdir
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class DirectXHeadersConan(ConanFile):
+    name = "directx-headers"
+    description = "Headers for using D3D12"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/DirectX-Headers"
+    topics = ("3d", "d3d", "d3d12", "direct", "direct3d", "directx", "graphics")
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "10",
+            "clang": "5",
+            "gcc": "6",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if not self.settings.os in ["Linux", "Windows"]:
+            raise ConanInvalidConfiguration(f"{self.name} is not supported on {self.settings.os}")
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.2.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = MesonToolchain(self)
+        tc.project_options["build-test"] = False
+        tc.generate()
+        virtual_build_env = VirtualBuildEnv(self)
+        virtual_build_env.generate()
+
+    def build(self):
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
+        meson.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        if self.settings.os == "Linux" or self.settings.get_safe("os.subsystem") == "wsl":
+            self.cpp_info.includedirs.append(os.path.join("include", "wsl", "stubs"))
```

Comment:
Why is this needed? By the project's README, it seems like the while `include/wsl` dir is useful in that configuration. Why just the `stubs` folder? (I might just be missing something really obvious upstream, please do point me at it if so!)

Reply:
Aw, so when not in WSL, there's a bunch of stuff provided by the Windows headers that aren't available on Linux. Since they are fixed system header files, they need to be found using the same include path. That `stubs` directory is specifically meant for being added as an include directory on Linux, in order to find the headers.

You can also see where they explicitly do this in the CMakeLists.txt [here](https://github.com/microsoft/DirectX-Headers/blob/c0f2af8af649833910786f38eb79cc333c77b1a6/CMakeLists.txt#L51-L55).

Reply:
Thank you. For some reason it didnt cross my mind to ctrl f the word stub in the cmakelist and I missed that line :|